src/http/server/handlers/deflate_handler.cr

```diff
@@ -0,0 +1,33 @@
+require "zlib"
+
+class HTTP::DeflateHandler < HTTP::Handler
+  DEFLATE_TYPES = %w(text/html text/plain text/xml text/css text/javascript application/javascript)
```

Comment:
@ysbaddaden Yeap. I grabbed the list from [mod_deflate](http://httpd.apache.org/docs/2.2/mod/mod_deflate.html#recommended)

@luislavena 
I think there should be a default deflate_types so some binary are skipped by default, instead of the non-set list as Rack::Deflater do.

Also I like the no-transform/identity check. And the Content-Type check with respect params after a `;`. I will probably add those too.